ldb/operational_test.go

```diff
@@ -39,24 +33,167 @@ func testOperationalMode(t *testing.T) {
+		return
+	}
+
+	// Extract the dest addr from the tx.
+	_, testAddrs, _, err := btcscript.ExtractPkScriptAddrs(testTx.MsgTx().TxOut[0].PkScript, &btcnet.MainNetParams)
+	if err != nil {
+		t.Errorf("Unable to decode tx output, err %v", err)
+		return
+	}
+
+	// Extract the hash160 from the output script.
+	var hash160Bytes [ripemd160.Size]byte
+	testHash160 := testAddrs[0].(*btcutil.AddressPubKey).AddressPubKeyHash().ScriptAddress()
+	copy(hash160Bytes[:], testHash160[:])
+
+	// Create a fake index.
+	blktxLoc, _ := newestBlock.TxLoc()
+	testIndex[hash160Bytes] = []*btcwire.TxLoc{&blktxLoc[0]}
+
+	// Insert our test addr index into the DB.
+	err = db.UpdateAddrIndexForBlock(newestSha, newestBlockIdx, &testIndex)
+	if err != nil {
+		t.Errorf("UpdateAddrIndexForBlock: failed to index"+
+			" addrs for block #%d (%s) "+
+			"err %v", newestBlockIdx, newestSha, err)
+		return
+	}
+
+	// Chain Tip of address should've been updated.
+	assertAddrIndexTipIsUpdated(db, t, newestSha, newestBlockIdx)
+
+	// Check index retrieval.
+	txReplies, err := db.FetchTxsForAddr(testAddrs[0], 0, 1000)
+	if err != nil {
+		t.Errorf("FetchTxsForAddr failed to correctly fetch txs for an "+
+			"address, err %v", err)
+		return
+	}
+	// Should have one reply.
+	if len(txReplies) == 0 {
+		t.Errorf("Failed to properly index tx by address.")
+		return
+	}
+
+	// Our test tx and indexed tx should have the same sha.
+	indexedTx := txReplies[0]
+	if !bytes.Equal(indexedTx.Sha.Bytes(), testTx.Sha().Bytes()) {
+		t.Errorf("Failed to fetch proper indexed tx. Expected sha %v, "+
+			"fetched %v", indexedTx.Sha, testTx.Sha())
+		return
+	}
+
+	// Shut down DB.
+	db.Sync()
+	db.Close()
+
+	// Re-Open, tip still should be updated to current height and sha.
+	db, err = btcdb.OpenDB("leveldb", "tstdbop1")
+	if err != nil {
+		t.Errorf("Unable to re-open created db, err %v", err)
+		return
+	}
+	assertAddrIndexTipIsUpdated(db, t, newestSha, newestBlockIdx)
+
+	// Delete the entire index.
+	err = db.DeleteAddrIndex()
+	if err != nil {
+		t.Errorf("Couldn't delete address index, err %v", err)
+		return
+	}
+
+	// Former index should no longer exist.
+	txReplies, err = db.FetchTxsForAddr(testAddrs[0], 0, 1000)
+	if len(txReplies) != 0 {
+		t.Errorf("Address index was not successfully deleted. "+
+			"Should have 0 tx's indexed, %v were returned.",
+			len(txReplies))
+		return
+	}
+
+	// Tip should be blanked out.
+	if _, _, err := db.FetchAddrIndexTip(); err != btcdb.ErrAddrIndexDoesNotExist {
+		t.Errorf("Address index was not fully deleted.")
+	}
+
+}
+
+func assertAddrIndexTipIsUpdated(db btcdb.Db, t *testing.T, newestSha *btcwire.ShaHash, newestBlockIdx int64) {
+	// Safe to ignore error, since height will be < 0 in "error" case.
+	sha, height, _ := db.FetchAddrIndexTip()
+	if newestBlockIdx != height {
+		t.Errorf("Height of address index tip failed to update, "+
+			"expected %v, got %v", newestBlockIdx, height)
+		return
```

Comment:
Agreed, had overlooked that function. Changed all instances of this pattern to use `t.Fatalf` instead.